everviz/pages/deltaplot.py

```diff
@@ -0,0 +1,78 @@
+import os
+from collections import namedtuple
+import pandas as pd
+from everviz.log import get_logger
+
+
+DataSources = namedtuple(
+    "DataSource", ["objective_delta_values", "summary_delta_values"]
+)
+
+logger = get_logger()
+
+
+def _get_objective_delta_values(api, best_batch):
+    data = pd.DataFrame(api.objective_values).drop(columns=["simulation"])
+    initial = data[(data.batch == 0)].drop(columns=["batch"])
+    best = data[data.batch == best_batch].drop(columns=["batch"])
+    initial = initial.pivot(index="realization", values="value", columns="function")
+    best = best.pivot(index="realization", values="value", columns="function")
+    return (best - initial).reset_index()
+
+
+def _get_summary_delta_values(api, best_batch):
+    data = (
+        api.summary_values()
+        .rename(columns={"simulation": "realization"})
+        .set_index(["realization", "date"])
+    )
+    initial = data[(data.batch == 0)].drop(columns=["batch"])
+    best = data[data.batch == best_batch].drop(columns=["batch"])
+    return (best - initial).reset_index()
+
+
+def _set_up_data_sources(api):
+    everest_folder = api.output_folder
+    everviz_path = os.path.join(everest_folder, "everviz")
+
+    objectives_df = pd.DataFrame(api.single_objective_values)
+    best_batch = objectives_df.batch[objectives_df.value.idxmax()]
+
+    logger.info("Generating objective delta plot source data file")
+    objective_csv_path = os.path.join(everviz_path, "objective_delta_values.csv")
+    data = _get_objective_delta_values(api, best_batch)
+    data.to_csv(objective_csv_path, index=False)
+    logger.info(f"File created: {objective_csv_path}")  # pylint: disable=W1203
+
+    logger.info("Generating summary delta plot source data file")
+    summary_csv_path = os.path.join(everviz_path, "summary_delta_values.csv")
+    data = _get_summary_delta_values(api, best_batch)
+    data.to_csv(summary_csv_path, index=False)
+    logger.info(f"File created: {summary_csv_path}")  # pylint: disable=W1203
+
+    return DataSources(
+        objective_delta_values=objective_csv_path, summary_delta_values=summary_csv_path
+    )
+
+
+def page_layout(api):
+    sources = _set_up_data_sources(api)
+    return {
+        "title": "Objectives Delta Values",
+        "content": [
+            "## Objective functions: Difference between best and initial batch",
+            {
+                "DeltaPlot": {
+                    "csv_file": sources.objective_delta_values,
+                    "pre_select": "all",
+                }
+            },
+            "## Summary keys: Difference between best and initial batch",
+            {
+                "DeltaPlot": {
+                    "csv_file": sources.summary_delta_values,
+                    "pre_select": "none",
```

Comment:
Is this needed ?

Reply:
This is the second plot for summary values, the DeltaPlot object is used twice on this page.

Reply:
Yes, I got that, my point was that I don't think you need to have the `pre_select` defined to `none` or defined at all, as far as I understand from the code it can be `xyz` instead of `none` and it would have the same result.

Reply:
Good point, you could fill in anything and maybe get unexpected behavior. I will change the code explicitly require the option and raise an error on unknown values.